kernel/arch/dreamcast/include/dc/pvr/pvr_prim.h

```diff
@@ -0,0 +1,359 @@
+    \brief                           Compiled headers for polygons and sprites
+    \ingroup pvr_primitives
+
+    @{
+*/
+
+/** \defgroup pvr_commands          TA Command Values
+    \brief                          Command values for submitting data to the TA
+    \ingroup                        pvr_primitives_headers
+
+    These are are appropriate values for TA commands. Use whatever goes with the
+    primitive type you're using.
+
+    @{
+*/
+#define PVR_CMD_POLYHDR     0x80840000  /**< \brief PVR polygon header. */
+#define PVR_CMD_VERTEX      0xe0000000  /**< \brief PVR vertex data */
+#define PVR_CMD_VERTEX_EOL  0xf0000000  /**< \brief PVR vertex, end of strip */
+#define PVR_CMD_USERCLIP    0x20000000  /**< \brief PVR user clipping area */
+#define PVR_CMD_MODIFIER    0x80000000  /**< \brief PVR modifier volume */
+#define PVR_CMD_SPRITE      0xA0000000  /**< \brief PVR sprite header */
+/** @} */
+
+/** \defgroup pvr_bitmasks          Constants and Masks
+    \brief                          Polygon header constants and masks
+    \ingroup                        pvr_primitives_headers
+
+    \note 
+    Thanks to the arrangement of constants, this is mainly a matter of
+    bit shifting to compile headers...
+
+    @{
+*/
+#define PVR_TA_CMD_TYPE            GENMASK(26, 24)
+#define PVR_TA_CMD_USERCLIP        GENMASK(17, 16)
+#define PVR_TA_CMD_MODIFIER        BIT(7)
+#define PVR_TA_CMD_MODIFIERMODE    BIT(6)
+#define PVR_TA_CMD_CLRFMT          GENMASK(5, 4)
+#define PVR_TA_CMD_TXRENABLE       BIT(3)
+#define PVR_TA_CMD_SPECULAR        BIT(2)
+#define PVR_TA_CMD_SHADE           BIT(1)
+#define PVR_TA_CMD_UVFMT           BIT(0)
+#define PVR_TA_PM1_DEPTHCMP        GENMASK(31, 29)
+#define PVR_TA_PM1_CULLING         GENMASK(28, 27)
+#define PVR_TA_PM1_DEPTHWRITE      BIT(26)
+#define PVR_TA_PM1_TXRENABLE       BIT(25)
+#define PVR_TA_PM1_MODIFIERINST    GENMASK(30, 29)
+#define PVR_TA_PM2_SRCBLEND        GENMASK(31, 29)
+#define PVR_TA_PM2_DSTBLEND        GENMASK(28, 26)
+#define PVR_TA_PM2_SRCENABLE       BIT(25)
+#define PVR_TA_PM2_DSTENABLE       BIT(24)
+#define PVR_TA_PM2_FOG             GENMASK(23, 22)
+#define PVR_TA_PM2_CLAMP           BIT(21)
+#define PVR_TA_PM2_ALPHA           BIT(20)
+#define PVR_TA_PM2_TXRALPHA        BIT(19)
+#define PVR_TA_PM2_UVFLIP          GENMASK(18, 17)
+#define PVR_TA_PM2_UVCLAMP         GENMASK(16, 15)
+#define PVR_TA_PM2_FILTER          GENMASK(14, 12)
+#define PVR_TA_PM2_MIPBIAS         GENMASK(11, 8)
+#define PVR_TA_PM2_TXRENV          GENMASK(7, 6)
+#define PVR_TA_PM2_USIZE           GENMASK(5, 3)
+#define PVR_TA_PM2_VSIZE           GENMASK(2, 0)
+#define PVR_TA_PM3_MIPMAP          BIT(31)
+#define PVR_TA_PM3_TXRFMT          GENMASK(30, 21)
+/** @} */
+
+/** PVR polygon header.
+
+    This is the hardware equivalent of a rendering context; you'll create one of
+    these from your pvr_poly_cxt_t and use it for submission to the hardware.
+*/
+typedef struct pvr_poly_hdr {
+alignas(32)
```

Comment:
`alignas(32)` is not... aligned?

Reply:
I don't know why, in my head it looked better like that, but now that you mention it, it actually looks like shit to me, so... Reverting. lol.